test/helpers/FakeIpcProvider.js

```diff
@@ -67,6 +68,10 @@ FakeIpcProvider.prototype.on = function (type, callback) {
     }
 };
 
+FakeIpcProvider.prototype.reset = function (type, callback) {
+    this.notificationCallbacks = [];
```

Comment:
Shouldn't it also reset the ``validation``, ``error``, and ``response`` array as well? and have you tested if this also works with the existing ``FakeIpcProvider2`` class? :)

Reply:
What would be gained by doing that?

Reply:
It would really reset the provider with the anyways no longer used injected items.

Reply:
I guess, but it's a fake provider - it's seems like its only purpose is to exercise what you want to test and there aren't failing tests for those properties.

Reply:
> but it's a fake provider 

Yes, it is. It is just the correct way to clean anything up. Who knows if this has an impact once on our test cases.

Reply:
Do you mean the lack of realism could have an impact in the future?  Idk - I think this is fine as it is tbh and answers the question in #3303.

Reply:
Yes, it could. Alone the wordings we are using do show that we think/thought and do not know. That's why we should implement the ``reset`` method as I was recommending above.

Reply:
> Alone the wordings we are using do show that we think/thought and do not know. 

Could you clarify this? Do you mean you're uncertain that the reason the tests pass is because the reset is mocked in now?

Reply:
No, Chris, all I mean is if we implement this reset method than in the correct way and not just what we think is enough for now. Could you please just do it.

Reply:
I don't think it's useful or necessary at the moment. If you think it is, add it though.

Reply:
Next time will I directly do it and just merge it instead of discussing it with you in such a childish way :)

Reply:
> Shouldn't it also reset the validation, error, and response array as well?

The answer would have been: "no this isn't possible because many test cases will fail". :)